### PR TITLE
Fix several bugs in servicemonitor

### LINKED ISSUE #688: Helm chart 2.0.0 invalid servicemonitor.yaml
The servicemonitor.yaml template contains multiple errors:

Here is the working version:

```yaml
{{- if .Values.metrics.serviceMonitor.enabled }}
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: {{ include "sealed-secrets.fullname" . }}
  {{- if .Values.metrics.serviceMonitor.namespace }}
  namespace: {{ .Values.metrics.serviceMonitor.namespace }}
  {{- else }}
  namespace: {{ include "sealed-secrets.namespace" . }}
  {{- end }}
  labels: {{- include "sealed-secrets.labels" . | nindent 4 }}
    {{- if .Values.metrics.serviceMonitor.labels }}
    {{- include "sealed-secrets.render" ( dict "value" .Values.metrics.serviceMonitor.labels "context" $) | nindent 4 }}
    {{- end }}
spec:
  endpoints:
    - honorLabels: true
      port: "{{ .Values.service.port }}"
      {{- if .Values.metrics.serviceMonitor.interval }}
      interval: {{ .Values.metrics.serviceMonitor.interval }}
      {{- end }}
      {{- if .Values.metrics.serviceMonitor.scrapeTimeout }}
      scrapeTimeout: {{ .Values.metrics.serviceMonitor.scrapeTimeout }}
      {{- end }}
      {{- if .Values.metrics.serviceMonitor.metricRelabelings }}
      metricRelabelings: {{ toYaml .Values.metrics.serviceMonitor.metricRelabelings | nindent 8 }}
      {{- end }}
      {{- if .Values.metrics.serviceMonitor.relabelings }}
      relabelings: {{ toYaml .Values.metrics.serviceMonitor.relabelings | nindent 8 }}
      {{- end }}
  namespaceSelector:
    matchNames:
      - {{ include "sealed-secrets.namespace" . }}
  selector:
    matchLabels: {{- include "sealed-secrets.matchLabels" . | nindent 6 }}
{{- end }}
```

### PR DESCRIPTION
**Pull request by alvneiayu**
**Description of the change**

After we did the major release, the chart has several issues inside the service monitor. This patch fixes them.

**Applicable issues**

- fixes #688

### CODE REVIEW

**Review by juan131 — APPROVED** [review]
LGTM

**Review by agarcia-oss — APPROVED** [review]
Go for it!

File: helm/sealed-secrets/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -3,19 +3,19 @@ apiVersion: monitoring.coreos.com/v1
...
-  namespace: {{ .Values.serviceMonitor.namespace }}
+  {{- if .Values.metrics.serviceMonitor.namespace }}
+  namespace: {{ .Values.metrics.serviceMonitor.namespace }}
   {{- else }}
   namespace: {{ include "sealed-secrets.namespace" . }}
   {{- end }}
   labels: {{- include "sealed-secrets.labels" . | nindent 4 }}
-    {{- if .Values.serviceMonitor.labels }}
-    {{- include "sealed-secrets.render" ( dict "value" .Values.serviceMonitor.labels "context" $) | nindent 4 }}
+    {{- if .Values.metrics.serviceMonitor.labels }}
```
Comment (jfrancisco0) [inline helm/sealed-secrets/templates/servicemonitor.yaml:12]: This is not documented in `values.yaml` or the README :)

Comment (sathieu) [reply helm/sealed-secrets/templates/servicemonitor.yaml:12]: Done in https://github.com/bitnami-labs/sealed-secrets/pull/635.

File: helm/sealed-secrets/templates/service.yaml [hunk 2]
```diff
@@ -11,7 +11,8 @@ metadata:
 spec:
   type: {{ .Values.service.type }}
   ports:
-    - port: {{ .Values.service.port }}
+    - name: http
+      port: {{ .Values.service.port }}
```
Comment (sathieu) [inline helm/sealed-secrets/templates/service.yaml:15]: This breaks `kubeseal` until https://github.com/bitnami-labs/sealed-secrets/pull/648 is merged.

**Pull request merged**